src/main/java/org/squbs/sample/SampleActor.java

```diff
@@ -0,0 +1,63 @@
+package org.squbs.sample;
+
+import akka.actor.AbstractActor;
+import akka.actor.ActorRef;
+import akka.actor.Cancellable;
+import akka.japi.pf.ReceiveBuilder;
+import scala.PartialFunction;
+import scala.runtime.BoxedUnit;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+import static org.squbs.sample.AppConstants.chunkEnd;
+import static org.squbs.sample.AppConstants.emptyRequest;
+
+/**
+ * This is the actor that handles the request messages.
+ */
+public class SampleActor extends AbstractActor {
+
+    public SampleActor() {
+        receive(ReceiveBuilder.
+            match(PingRequest.class, r -> {
+                if (r.who.trim().length() > 0) {
+                    sender().tell(new PingResponse("Hello " + r.who + " welcome to squbs!"), self());
+                } else {
+                    sender().tell(emptyRequest, self());
+                }
+                context().stop(self());
+            }).
+            match(ChunkRequest.class, r -> {
+                ActorRef requester = sender(); // Save the requester for use in the scheduler.
+                String[] responseArray = {"Hello ", r.who, " welcome ", "to ", "squbs!"};
+                Stream<String> responseStream = Arrays.stream(responseArray);
+                if (r.delay.toMillis() > 0) {
+                    Iterator<String> responses = responseStream.iterator();
+                    Cancellable scheduler = context().system().scheduler().schedule(r.delay, r.delay, () -> {
```

Comment:
We should build a stream here and send the stream definition to the `sender()` instead of sending message by message to the `sender`.  Please see https://github.com/paypal/activator-squbs-scala/blob/akka-http/src/main/scala/org/squbs/sample/SampleActor.scala#L43.